src/Compilers/CSharp/Portable/Syntax/SyntaxExtensions.cs

```diff
@@ -228,6 +234,38 @@ private static bool IsInContextWhichNeedsDynamicAttribute(CSharpSyntaxNode node)
             }
         }
 
+        private static bool IsInContextWhichNeedsTupleNamesAttribute(CSharpSyntaxNode node)
+        {
+            Debug.Assert(node != null);
+
+            switch (node.Kind())
+            {
+                case SyntaxKind.Parameter:
+                case SyntaxKind.FieldDeclaration:
+                case SyntaxKind.MethodDeclaration:
+                case SyntaxKind.IndexerDeclaration:
+                case SyntaxKind.OperatorDeclaration:
+                case SyntaxKind.ConversionOperatorDeclaration:
+                case SyntaxKind.PropertyDeclaration:
+                case SyntaxKind.DelegateDeclaration:
+                case SyntaxKind.EventDeclaration:
+                case SyntaxKind.EventFieldDeclaration:
+                case SyntaxKind.BaseList:
+                case SyntaxKind.SimpleBaseType:
+                case SyntaxKind.TypeParameterConstraintClause:
+                    return true;
+
+                case SyntaxKind.Block:
+                case SyntaxKind.VariableDeclarator:
+                case SyntaxKind.Attribute:
+                case SyntaxKind.EqualsValueClause:
+                    return false;
+
+                default:
+                    return node.Parent != null && IsInContextWhichNeedsDynamicAttribute(node.Parent);
```

Comment:
perhaps use "?."  

---
In reply to: [69636880](https://github.com/dotnet/roslyn/pull/12344#discussion_r69636880) [](ancestors = 69636880)